File: $_algotsu-kaizen/problemo-set/maximize-happiness-of-selected-children.rs
impl Solution {
    pub fn maximum_happiness_sum(happiness: Vec<i32>, k: i32) -> i64 {
        let mut happiness = happiness;
        let k = k as usize;

        let mut result = 0x0i64;
        happiness.sort_by(|a, b| b.cmp(a));

        for (i, &h) in happiness.iter().enumerate() {
            if i >= k {
                break;
            }

            let kai = (h as i64) - (i as i64);
            if kai <= 0x0 {
                break;
            }

            result += kai;
        }

        result
    }
}

File: $_algotsu-kaizen/problemo-set/maximize-happiness-of-selected-children.c
int compare(const void *a, const void *b) {
    int A = *(int*)a;
    int B = *(int*)b;
    return B - A;
}

long long maximumHappinessSum(int* happiness, int happinessSize, int k) {
    qsort(happiness, happinessSize, sizeof(int), compare);

    long long result = 0x0;

    for (int i = 0x0; i < k; ++i) {
        int kai = happiness[i] - i;
        if (kai <= 0x0) break;
        result += kai;
    }

    return result;
}
